plugins/modules/cm_import_cluster_template.py

```diff
@@ -0,0 +1,198 @@
+    port: "7180"
+    template: "./files/cluster-template.json"
+    add_repositories: "True"
+"""
+
+RETURN = r"""
+---
+cloudera_manager:
+    description: Details about Cloudera Manager Cluster
+    type: dict
+    contains:
+        cluster_type:
+            description: The type of cluster created from template.
+            type: str
+            returned: optional
+        cluster_url:
+            description: Url of Cloudera Manager cluster.
+            type: str
+            returned: optional
+        display_name:
+            description: The name of the cluster displayed on the site.
+            type: str
+            returned: optional
+        entity_status:
+            description: Health status of the cluster.
+            type: str
+            returned: optional
+        full_version:
+            description: Version of the cluster installed.
+            type: str
+            returned: optional
+        hosts_url:
+            description: Url of all the hosts on which cluster is installed.
+            type: str
+            returned: optional
+        maintenance_mode:
+            description: Maintance mode of Cloudera Manager Cluster.
+            type: bool
+            returned: optional
+        maintenance_owners:
+            description: List of Maintance owners for Cloudera Manager Cluster.
+            type: list
+            returned: optional
+        name:
+            description: The name of the cluster created.
+            type: str
+            returned: optional
+        tags:
+            description: List of tags for Cloudera Manager Cluster.
+            type: list
+            returned: optional
+        uuid:
+            description: Unique ID of created cluster
+            type: bool
+            returned: optional
+"""
+
+
+class ClusterTemplate(ClouderaManagerModule):
+    def __init__(self, module):
+        super(ClusterTemplate, self).__init__(module)
+        self.template = self.get_param("template")
+        self.add_repositories = self.get_param("add_repositories")
+        self.process()
+
+    @ClouderaManagerModule.handle_process
+    def process(self):
+        
+        try:
+            api_instance = ClouderaManagerResourceApi(self.api_client)
+            cluster_api_instance = ClustersResourceApi(self.api_client)
+
+            with open(self.template, 'r') as file:
+                template_json = json.load(file)
+            if self.add_repositories:
+                import_template_request = api_instance.import_cluster_template(add_repositories=True,body=template_json).to_dict()
+            else:
+                import_template_request = api_instance.import_cluster_template(body=template_json).to_dict()
+
+            command_id = import_template_request['id']
+
+            self.wait_for_command_state(command_id=command_id,polling_interval=60)
+            
+            self.cm_cluster_template_output = cluster_api_instance.read_clusters().to_dict()
```

Comment:
Actually, this should call the `/clusters/{clustername}` and should call it initially to figure out if the activity is a _create_ or an _update_. (The latter should be split out as separate, follow-on PRs.)